dotnet/samples/08.datasource.azureaisearch/AzureAISearchBot/AzureAISearchDataSource.cs

```diff
@@ -0,0 +1,166 @@
+
+        public readonly AzureAISearchDataSourceOptions Options;
+
+        public readonly SearchClient SearchClient;
+
+        public AzureAISearchDataSource(AzureAISearchDataSourceOptions options)
+        {
+            Options = options;
+            Name = options.Name;
+
+            AzureKeyCredential credential = new AzureKeyCredential(options.AzureAISearchApiKey);
+            SearchClient = new SearchClient(options.AzureAISearchEndpoint, options.IndexName, credential);
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async Task<RenderedPromptSection<string>> RenderDataAsync(ITurnContext context, IMemory memory, ITokenizer tokenizer, int maxTokens, CancellationToken cancellationToken = default)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            string query = (string)memory.GetValue("temp.input")!;
+
+            if (string.IsNullOrEmpty(query))
+            {
+                return new RenderedPromptSection<string>("");
+            }
+
+            List<string> selectedFields = new() { "RestaurantId", "RestaurantName", "Description", "Category", "Tags", "Rating", "Address" };
+            List<string> searchFields = new() { "Category", "Tags", "RestaurantName" };
+
+            //// TEXT SEARCH ////
+            //// Get the restaurants with the query has high lexical relevance to category, tags, and restaurant name.
+            SearchOptions options = new();
+            foreach (string field in searchFields)
+            {
+                options.SearchFields.Add(field);
+            }
+
+            foreach (string field in selectedFields)
+            {
+                options.Select.Add(field);
+            }
+            SearchResults<Restaurant> search = SearchClient.Search<Restaurant>(query, options);
+
+            //// VECTOR SEARCH ////
+            //// Get the restaurants with description that is the most similar to the user input.
+            /*SearchOptions options = new();
+            ReadOnlyMemory<float> vectorizedQuery = await this._GetEmbeddingVector(query);
+            foreach (string field in selectedFields)
+            {
+                options.Select.Add(field);
+            }
+            options.VectorSearch = new()
+            {
+                Queries = { new VectorizedQuery(vectorizedQuery) { KNearestNeighborsCount = 3, Fields = { "DescriptionVectorEn" } } }
+            };
+            SearchResults<Restaurant> search = SearchClient.Search<Restaurant>(options);
+            */
+
+            //// HYBRID SEARCH ////
+            //// Search using both vector and text search
+            /*SearchOptions options = new();
+            ReadOnlyMemory<float> vectorizedQuery = await this._GetEmbeddingVector(query);
+            foreach (string field in searchFields)
+            {
+                options.SearchFields.Add(field);
+            }
```

Comment:
Is this all supposed to be commented out?

Reply:
yes users have 3 search type to choose from "simple", "vector" & "hybrid". Default is "simple"